src/server/qgswfsserver.cpp

```diff
@@ -445,7 +445,7 @@ int QgsWFSServer::getFeature( QgsRequestHandler& request, const QString& format
     QDomElement queryElem;
     for ( int i = 0; i < queryNodes.size(); i++ )
     {
-      queryElem = queryNodes.at( 0 ).toElement();
+      queryElem = queryNodes.at( i ).toElement();
```

Comment:
Hi, I know this is unrelated with your modifications, but if I'm not wrong, this first loop at line 446 is dead code: all what it does it setting `mTypeNames` that it's not used and it's being redefined around line 878.
Would you please have a look and remove the loop if it's dead?